File: apps/api-ts/src/app/products/products.service.ts
import { Injectable, Logger } from '@nestjs/common';
import { ProductsRepository } from './db/products.repository';
import { Product } from './db/schemas/product.schema';

@Injectable()
export class ProductsService {
    private readonly logger = new Logger(ProductsService.name);

    constructor(private readonly repository: ProductsRepository) {}

    async create(product: Product) {
        this.logger.log('Creating product');
        return this.repository.create(product);
    }
}

File: apps/api-java/src/main/java/com/example/demo/products/ProductsService.java
package com.example.demo.products;

import com.example.demo.products.db.Product;
import com.example.demo.products.db.ProductsRepository;
import org.springframework.stereotype.Service;

@Service
public class ProductsService {

  private final ProductsRepository repository;

  public ProductsService(final ProductsRepository repository) {
    this.repository = repository;
  }

  public Product create(Product product) {
    return repository.save(product);
  }
}
